src/main/java/org/testng/ITestRunnerFactory2.java

```diff
@@ -0,0 +1,28 @@
+package org.testng;
+
+import org.testng.xml.XmlTest;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A factory for TestRunners to be used by SuiteRunners.
+ */
+public interface ITestRunnerFactory2 extends ITestRunnerFactory {
```

Comment:
Ok but no need to have a default method. Because once the method will be deprecated (but still supported), the step after will be to remove it.

What I propose is using the new variant now and make a pattern tricks in order to support the "old" one.